pybtp/btp.py

```diff
@@ -3651,6 +3893,1158 @@ def mesh_lpn_polled_ev(mesh, data, data_len):
+
+    hdr_fmt = '<b'
+    on_power_up, = struct.unpack_from(hdr_fmt, rsp)
+    stack = get_stack()
+    stack.mesh.rcv_status_data_set('Status', [on_power_up])
+    logging.debug("Status on power up = %r", stack.mesh.rcv_status_data_get('Status'))
+
+def mmdl_gen_ponoff_set(on_power_up, ack=True):
+    logging.debug("%s", mmdl_gen_ponoff_set.__name__)
+
+    iutctl = get_iut()
+    data = bytearray(struct.pack("<BB", ack, on_power_up))
+
+    (rsp,) = iutctl.btp_socket.send_wait_rsp(*MMDL['gen_ponoff_set'], data=data)
+
+    if ack:
+        hdr_fmt = '<b'
+        on_power_up, = struct.unpack_from(hdr_fmt, rsp)
+        stack = get_stack()
+        stack.mesh.rcv_status_data_set('Status', [on_power_up])
+        logging.debug("Status on power up = %r", stack.mesh.rcv_status_data_get('Status'))
+
+def mmdl_gen_plvl_get():
+    logging.debug("%s", mmdl_gen_plvl_get.__name__)
+
+    iutctl = get_iut()
+    (rsp,) = iutctl.btp_socket.send_wait_rsp(*MMDL['gen_plvl_get'])
+
+    hdr_fmt = '<H'
+    power_level, = struct.unpack_from(hdr_fmt, rsp)
+    stack = get_stack()
+    stack.mesh.rcv_status_data_set('Status', [power_level])
+    logging.debug("Status power level = %r", stack.mesh.rcv_status_data_get('Status'))
+
+def mmdl_gen_plvl_set(power_lvl, tt=None, delay=None, ack=True):
+    logging.debug("%s", mmdl_gen_plvl_set.__name__)
+
+    iutctl = get_iut()
+    data = bytearray(struct.pack("<BH", ack, power_lvl))
+    if tt is not None:
+        data.extend(struct.pack("<B", tt))
+    if delay is not None:
+        data.extend(struct.pack("<B", delay))
+
+    (rsp,) = iutctl.btp_socket.send_wait_rsp(*MMDL['gen_plvl_set'], data=data)
+
+    if ack:
+        hdr_fmt = '<HHi'
+        present_power, target_power, remaining_time = struct.unpack_from(hdr_fmt, rsp)
+        stack = get_stack()
+        if remaining_time:
+            stack.mesh.rcv_status_data_set("Status", [target_power])
+        else:
+            stack.mesh.rcv_status_data_set('Status', [present_power])
+        logging.debug("Status power = %r", stack.mesh.rcv_status_data_get('Status'))
+
+def mmdl_gen_plvl_last_get():
+    logging.debug("%s", mmdl_gen_plvl_last_get.__name__)
+
+    iutctl = get_iut()
+    (rsp,) = iutctl.btp_socket.send_wait_rsp(*MMDL['gen_plvl_last_get'])
+
+    hdr_fmt = '<H'
+    power_last, = struct.unpack_from(hdr_fmt, rsp)
+    stack = get_stack()
+    stack.mesh.rcv_status_data_set('Status', [power_last])
+    logging.debug("Status power last = %r", stack.mesh.rcv_status_data_get('Status'))
+
+
+def mmdl_gen_plvl_dflt_get():
+    logging.debug("%s", mmdl_gen_plvl_dflt_get.__name__)
+
+    iutctl = get_iut()
+    (rsp,) = iutctl.btp_socket.send_wait_rsp(*MMDL['gen_plvl_dflt_get'])
+
+    hdr_fmt = '<H'
+    present_power, = struct.unpack_from(hdr_fmt, rsp)
+    stack = get_stack()
+    stack.mesh.rcv_status_data_set('Status', [present_power])
+    logging.debug("Status power = %r", stack.mesh.rcv_status_data_get('Status'))
+
```

Comment:
I don't remember what's the recommended number of empty lines between function definitions, but please be consistent. Here you have one empty line while in other places you put 2.